src/Analyzers/CSharp/Analyzers/ConvertNameOf/CSharpConvertNameOfDiagnosticAnalyzer.cs

```diff
@@ -80,5 +81,29 @@ private void AnalyzeSyntaxAction(SyntaxNodeAnalysisContext syntaxContext)
         public override DiagnosticAnalyzerCategory GetAnalyzerCategory()
             => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
 
+        private static bool ArgumentIsGeneric(IOperation op)
+        {
+            // Cast it to a ITypeOfOperation
+            var tOp = (ITypeOfOperation)op;
+
+            if (((INamedTypeSymbol)(tOp).TypeOperand).IsGenericType)
+            {
+                return true;
+            }
+            return false;
+        }
+
+        private static bool ArgumentIsPrimitive(IOperation op)
```

Comment:
A better way to check for primitive type is something like: https://github.com/dotnet/roslyn-analyzers/blob/f15404b312295e7cee16fb40b4c8d3f91f10f087/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DoNotLockOnObjectsWithWeakIdentity.cs#L124-L144

Maybe you can invoke the below method on `ITypeSymbol.SpecialType` and check if the returned predefined type is not `None`? https://github.com/dotnet/roslyn/blob/f4d410e519109d676f78bdb4352022350ce0e034/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/SpecialTypeExtensions.cs#L13